drivers/usb/udc/udc_common.c

```diff
@@ -0,0 +1,1112 @@
+
+	return USB_REQTYPE_GET_DIR(setup->bmRequestType);
+}
+
+static bool udc_data_stage_to_device(const struct net_buf *const buf)
+{
+	return !udc_data_stage_to_host(buf);
+}
+
+static ALWAYS_INLINE
+struct net_buf *udc_ctrl_alloc_stage(const struct device *dev,
+				     struct net_buf *const parent,
+				     const uint8_t ep,
+				     const size_t size)
+{
+	struct net_buf *buf;
+
+	buf = udc_ctrl_alloc(dev, ep, size);
+	if (buf == NULL) {
+		return NULL;
+	}
+
+	if (parent) {
+		net_buf_frag_add(parent, buf);
+	}
+
+	return buf;
+}
+
+static struct net_buf *udc_ctrl_alloc_data(const struct device *dev,
+					   struct net_buf *const setup,
+					   const uint8_t ep)
+{
+	size_t size = udc_data_stage_length(setup);
+	struct udc_buf_info *bi;
+	struct net_buf *buf;
+
+	buf = udc_ctrl_alloc_stage(dev, setup, ep, size);
+	if (buf) {
+		bi = udc_get_buf_info(buf);
+		bi->data = true;
+	}
+
+	return buf;
+}
+
+static struct net_buf *udc_ctrl_alloc_status(const struct device *dev,
+					     struct net_buf *const parent,
+					     const uint8_t ep)
+{
+	size_t size = (ep == USB_CONTROL_EP_OUT) ? 64 : 0;
+	struct udc_buf_info *bi;
+	struct net_buf *buf;
+
+	buf = udc_ctrl_alloc_stage(dev, parent, ep, size);
+	if (buf) {
+		bi = udc_get_buf_info(buf);
+		bi->status = true;
+	}
+
+	return buf;
+}
+
+int udc_ctrl_submit_s_out_status(const struct device *dev,
+			      struct net_buf *const dout)
+{
+	struct udc_buf_info *bi = udc_get_buf_info(dout);
+	struct udc_data *data = dev->data;
+	struct net_buf *buf;
+	int ret = 0;
+
+	bi->data = true;
+	net_buf_frag_add(data->setup, dout);
+
+	buf = udc_ctrl_alloc_status(dev, dout, USB_CONTROL_EP_IN);
+	if (buf == NULL) {
+		ret = -ENOMEM;
+	}
+
+	return udc_submit_event(dev, UDC_EVT_EP_REQUEST, ret, data->setup);
+}
+
+int udc_ctrl_submit_s_in_status(const struct device *dev)
+{
+	struct udc_data *data = dev->data;
+	struct net_buf *buf;
+	int ret = 0;
+
+	if (!udc_ctrl_stage_is_data_in(dev)) {
+		return -ENOTSUP;
+	}
+
+	/* Allocate buffer for data stage IN */
+	buf = udc_ctrl_alloc_data(dev, data->setup, USB_CONTROL_EP_IN);
+	if (buf == NULL) {
+		ret = -ENOMEM;
+	}
+
+	return udc_submit_event(dev, UDC_EVT_EP_REQUEST, ret, data->setup);
+}
+
+int udc_ctrl_submit_s_status(const struct device *dev)
+{
+	struct udc_data *data = dev->data;
+	struct net_buf *buf;
+	int ret = 0;
+
+	/* Allocate buffer for possible status IN */
+	buf = udc_ctrl_alloc_status(dev, data->setup, USB_CONTROL_EP_IN);
+	if (buf == NULL) {
+		ret = -ENOMEM;
+	}
+
+	return udc_submit_event(dev, UDC_EVT_EP_REQUEST, ret, data->setup);
+}
+
+int udc_ctrl_submit_status(const struct device *dev,
+			   struct net_buf *const buf)
+{
+	struct udc_buf_info *bi = udc_get_buf_info(buf);
+
+	bi->status = true;
+
+	return udc_submit_event(dev, UDC_EVT_EP_REQUEST, 0, buf);
+}
+
+bool udc_ctrl_stage_is_data_out(const struct device *dev)
+{
+	struct udc_data *data = dev->data;
+
+	return data->stage == CTRL_PIPE_STAGE_DATA_OUT ? true : false;
```

Comment:
Why still use the ternary operator?

```suggestion
	return data->stage == CTRL_PIPE_STAGE_DATA_OUT;
```

Reply:
Because the type `data->stage == CTRL_PIPE_STAGE_DATA_OUT` is an int not bool.